File: Expressions & Assignment/Destructuring.py
values = [10, 20, 30, 40, 50]
first, second, *rest = values

obj = {"name": "Carlos", "age": 22}
full_name = obj.get("name")
age = obj.get("age", 18)
city = obj.get("city", "Unknown")

print(first)
print(rest)
print(full_name)
print(city)

File: Expressions & Assignment/Destructuring.js
// Array destructuring
let [first, second, ...rest] = [10, 20, 30, 40, 50];

// Object destructuring with renaming and defaults
let { name: fullName, age = 18, city = "Unknown" } = { name: "Carlos", age: 22 };

console.log(first);    // 10
console.log(rest);     // [30, 40, 50]
console.log(fullName); // "Carlos"
console.log(city);     // "Unknown" (default)